tests/by-util/test_du.rs

```diff
@@ -1430,3 +1431,240 @@ fn test_du_inodes_total_text() {
+
+    let result = ts.ucmd().arg("-L").arg("shallow_link.txt").succeeds();
+    assert!(!result.stdout_str().is_empty());
+
+    let result = ts.ucmd().arg("shallow_link.txt").succeeds();
+    assert!(!result.stdout_str().is_empty());
+}
+#[test]
+#[cfg(target_os = "linux")]
+fn test_du_inaccessible_directory() {
+    // tested by tests/du/no-x
+    let ts = TestScenario::new(util_name!());
+    let at = ts.fixtures.clone();
+
+    at.mkdir("d");
+    at.mkdir("d/no-x");
+    at.mkdir("d/no-x/y");
+
+    at.set_mode("d/no-x", 0o600);
+
+    let result = ts.ucmd().arg("d").fails();
+    result.stderr_contains("du: cannot access 'd/no-x/y': Permission denied");
+}
+
+#[test]
+#[cfg(unix)]
+fn test_du_symlink_self_reference() {
+    // Test symlink that points to its own directory
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("selfref");
+    // Create symlink pointing to selfref directory itself: selfref/self -> selfref
+    at.symlink_dir("selfref", "selfref/self");
+
+    let result = ts.ucmd().arg("-L").arg("selfref").succeeds();
+
+    result.stdout_contains("selfref");
+    // Should not show the self-referencing symlink to avoid infinite recursion
+    result.stdout_does_not_contain("selfref/self");
+}
+
+#[test]
+#[cfg(unix)]
+fn test_du_long_symlink_chain() {
+    // Test that very long symlink chains are handled gracefully
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    // Create a simple structure that tests symlink depth limits
+    // Instead of trying to create a chain that causes ELOOP, test that reasonable chains work
+    at.mkdir_all("deep/level1/level2/level3/level4/level5");
+    at.write(
+        "deep/level1/level2/level3/level4/level5/file.txt",
+        "content",
+    );
+
+    at.symlink_dir("deep/level1", "link1");
+    at.symlink_dir("link1/level2", "link2");
+    at.symlink_dir("link2/level3", "link3");
+
+    let result = ts.ucmd().arg("-L").arg("link3").succeeds();
+    result.stdout_contains("link3");
+}
+
+#[test]
+#[cfg(all(unix, not(target_os = "macos")))]
+fn test_du_bind_mount_simulation() {
+    // Simulate bind mount scenario using hard links where possible
+    // Note: This test simulates what bind mounts do - making the same directory
+    // appear in multiple places with the same inode
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir_all("mount_test/subdir");
+    at.write("mount_test/file1.txt", "content1");
+    at.write("mount_test/subdir/file2.txt", "content2");
+
+    // On systems where we can't create actual bind mounts,
+    // we test that cycle detection works with symlinks that would create similar cycles
+    at.symlink_dir("../mount_test", "mount_test/subdir/cycle_link");
+
+    let result = ts.ucmd().arg("mount_test").succeeds();
+
+    result.stdout_contains("mount_test/subdir");
+    result.stdout_contains("mount_test");
```

Comment:
The second check currently can never fail. So I would either swap the two lines or remove the second check.

Reply:
removed, it isn't interesting